pokemongo_bot/base_task.py

```diff
@@ -1,31 +1,43 @@
 import logging
+import time
 
 
 class BaseTask(object):
-  TASK_API_VERSION = 1
-
-  def __init__(self, bot, config):
-    self.bot = bot
-    self.config = config
-    self._validate_work_exists()
-    self.logger = logging.getLogger(type(self).__name__)
-    self.initialize()
-
-  def _validate_work_exists(self):
-    method = getattr(self, 'work', None)
-    if not method or not callable(method):
-      raise NotImplementedError('Missing "work" method')
-
-  def emit_event(self, event, sender=None, level='info', formatted='', data={}):
-    if not sender:
-      sender=self
-    self.bot.event_manager.emit(
-      event,
-      sender=sender,
-      level=level,
-      formatted=formatted,
-      data=data
-    )
-
-  def initialize(self):
-    pass
+    TASK_API_VERSION = 1
+
+    def __init__(self, bot, config):
+        self.bot = bot
+        self.config = config
+        self._validate_work_exists()
+        self.logger = logging.getLogger(type(self).__name__)
+        self.last_ran = time.time()
+        self.run_interval = config.get('run_interval', 10)
```

Comment:
@TheSavior the default value is 10 to make it work it the task does run in a time-based way. Otherwise if task is time-based and user don't provide `run_interval` he will see an error.